scan-payment/src/main/java/com/getbouncer/scan/payment/ml/ExpiryDetect.kt

```diff
@@ -0,0 +1,136 @@
+package com.getbouncer.scan.payment.ml
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.util.Size
+import com.getbouncer.scan.framework.Loader
+import com.getbouncer.scan.framework.ResourceLoader
+import com.getbouncer.scan.framework.ml.TFLAnalyzerFactory
+import com.getbouncer.scan.framework.ml.TensorFlowLiteAnalyzer
+import com.getbouncer.scan.framework.util.indexOfMax
+import com.getbouncer.scan.payment.R
+import com.getbouncer.scan.payment.hasOpenGl31
+import com.getbouncer.scan.payment.scale
+import com.getbouncer.scan.payment.toRGBByteBuffer
+import org.tensorflow.lite.Interpreter
+import java.io.FileNotFoundException
+import java.nio.ByteBuffer
+
+private val TRAINED_IMAGE_SIZE = Size(80, 36)
+
+private const val NUM_CLASS = 11
+private const val NUM_PREDICTIONS = 17
+private const val BACKGROUND_CLASS = 10
+
+val ASPECT_RATIO = TRAINED_IMAGE_SIZE.height.toFloat() / TRAINED_IMAGE_SIZE.width.toFloat()
+
+class ExpiryDetect private constructor(interpreter: Interpreter) :
+    TensorFlowLiteAnalyzer<ExpiryDetect.Input, ByteBuffer,
+        ExpiryDetect.Prediction,
+        Array<Array<Array<FloatArray>>>>(interpreter) {
+
+    data class Input(val image: Bitmap)
+
+    data class Prediction(val expiry: Expiry?)
+
+    data class Expiry(val month: Int, val year: Int) {
+        override fun toString() = "${month.toString().padStart(2, '0')}/$year"
```

Comment:
is `year` guaranteed to always be 2 or 4 digits? should it also have a `padStart`?

Reply:
yeah, always guaranteed to be 4 digits